File: Lv.1/소수 찾기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#include <math.h>
#include <functional>
using namespace std;

int solution(int n) {
    int answer = 0;
    
    // 에라토스테네스의 채 이용
    bool isPrime[1000001];
    for(int i=0;i<1000001;i++){
        isPrime[i]=true;
    }
    
    isPrime[0]=false;
    isPrime[1]=false;
    for(int i=2;i<=sqrt(1000001);i++){
        if(isPrime[i]==true){
            for(int j=2*i;j<1000001;j+=i){
                isPrime[j]=false;
            }
        }
    }
    
    for(int i=1;i<=n;i++){
        if(isPrime[i]==true){
            answer++;
        }
    }
    return answer;
}

File: Java/LV1/소수 찾기.java
import java.util.*;
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        boolean[] arr = new boolean[n+1];
        for(int i=0;i<arr.length;i++){
            arr[i]=true;
        }
        arr[0]=false; // 소수 아님
        arr[1]=false; // 소수 아님
        
        for(int i=2;i*i<=n;i++){
            if(arr[i]==true){
                for(int j=i+i;j<=n;j+=i){
                    arr[j]=false;
                }
            }
  
        }
        
        for(int i=1;i<=n;i++){
            if(arr[i]==true){
                answer++;
            }
        }
        return answer;
    }
}
